commands.go

```diff
@@ -12,466 +12,412 @@ import (
-	devbotChat(u.room, line, sendToSlack)
-	//}
+	if strings.HasPrefix(args[0], "#") {
+		// It's a normal room
 
-	if strings.HasPrefix(line, "./tic") {
-		rest := strings.TrimSpace(strings.TrimPrefix(line, "./tic"))
-		if rest == "" {
-			b(devbot, "Starting a new game of Tic Tac Toe! The first player is always X.")
-			b(devbot, "Play using ./tic <cell num>")
-			currentPlayer = tictactoe.X
-			tttGame = new(tictactoe.Board)
-			b(devbot, "```\n"+" 1 │ 2 │ 3\n───┼───┼───\n 4 │ 5 │ 6\n───┼───┼───\n 7 │ 8 │ 9\n"+"\n```")
+		roomName := strings.TrimPrefix(args[0], "#")
+		if len(roomName) == 0 {
+			u.system("You need to give me a channel name to move you to!")
 			return
 		}
-
-		m, err := strconv.Atoi(rest)
-		if err != nil {
-			b(devbot, "Make sure you're using a number lol")
-			return
+		newRoom, exists := rooms[roomName]
+		if !exists {
+			newRoom = &room{roomName, make([]*user, 0, 10), sync.Mutex{}}
+			rooms[roomName] = newRoom
 		}
-		if m < 1 || m > 9 {
-			b(devbot, "Moves are numbers between 1 and 9!")
+		u.changeRoom(newRoom, true)
+		u.system(fmt.Sprintf("Moved you to %s", roomName))
+		return
+	}
+	if strings.HasPrefix(args[0], "@") {
+		userName := strings.TrimPrefix(args[0], "@")
+		if len(userName) == 0 {
+			u.system("You have to tell me who you want to message")
 			return
 		}
-		err = tttGame.Apply(tictactoe.Move(m-1), currentPlayer)
-		if err != nil {
-			b(devbot, err.Error())
+		peer, ok := findUserByName(u.room, userName)
+		if !ok {
+			u.system("No person in your room found with that name")
 			return
 		}
-		b(devbot, "```\n"+tttPrint(tttGame.Cells)+"\n```")
-		if currentPlayer == tictactoe.X {
-			currentPlayer = tictactoe.O
-		} else {
-			currentPlayer = tictactoe.X
-		}
-		if !(tttGame.Condition() == tictactoe.NotEnd) {
-			b(devbot, tttGame.Condition().String())
-			currentPlayer = tictactoe.X
-			tttGame = new(tictactoe.Board)
-			// hasStartedGame = false
-		}
+		u.messaging = peer
+		u.system(fmt.Sprintf("Now messaging %s. To leave use ./room leave", u.messaging.name))
 		return
 	}
+	u.system("Invalid usage. Valid usage: ./room leave|#room-name|@user-name")
+}
 
-	if line == "./users" {
-		b("", printUsersInRoom(u.room))
+func kickCommand(u *user, args []string) {
+	if len(args) != 1 {
+		u.system("Please provide a user to kick!")
 		return
 	}
-	if line == "./all" {
-		names := make([]string, 0, len(allUsers))
-		for _, name := range allUsers {
-			names = append(names, name)
-		}
-		sort.Slice(names, func(i, j int) bool {
-			return strings.ToLower(stripansi.Strip(names[i])) < strings.ToLower(stripansi.Strip(names[j]))
-		})
-		b("", fmt.Sprint(names))
+	target, ok := findUserByName(u.room, args[0])
+	if !ok {
+		u.system("User not found!")
 		return
 	}
-	if line == "easter" {
-		go func() {
-			time.Sleep(time.Second)
-			b(devbot, "eggs?")
-		}()
+	target.system(fmt.Sprintf("You have been kicked by %s", u.name))
+	target.close(fmt.Sprintf(red.Paint("%s was kicked by %s"), target.name, u.name))
```

Comment:
make close automatically send a system message, instead of repeating the message with u.system